src/awkward/forms/form.py

```diff
@@ -427,6 +428,12 @@ def _type(self, typestrs):
     def length_zero_array(
         self, *, backend=numpy_backend, highlevel=True, behavior=None
     ):
+        if highlevel:
```

Comment:
It's correct to trigger the warning on `highlevel` (ignoring `behavior`) because `behavior` has no effect when `highlevel is False`.